pkg/models/dashboards.go

```diff
@@ -91,8 +92,11 @@ func NewDashboardFromJson(data map[string]interface{}) *Dashboard {
 // GetDashboardModel turns the command into the savable model
 func (cmd *SaveDashboardCommand) GetDashboardModel() *Dashboard {
 	dash := NewDashboardFromJson(cmd.Dashboard)
+	if dash.Data["version"] == 0 {
```

Comment:
Actually, we should not use dash.Version here because dash.Version is not set properly until this line of code. Anyway, I tried using dash.Version as suggested by you but then saving dashboard is updating "created_by" field too in the DB while saving the dashboard again, which is incorrect. Also, If you look at line 83 in the same file, you will find that we use the same if condition to create the dashboard object. So, I think lets stick with this if condition and not change it for the correct functionality. :)